### DISCUSSION TITLE
Run Stryket on a solution with multiple test projects

### DISCUSSION THREAD

**Discussion by CarlosEGSouza** [stryker-mutator/stryker-net#2815] [Q&A] [answered]
Hello,

In my solution i have three test projects (Unit, UI and Integration). How do i run stryker only in a unit test project? (C# net6.0 in VS 2002)

I created a **stryker-config.json** in the solution path like this, but **aways** stryker runs ALL tests from all three projects. Why??

```
{
    "stryker-config":
    {
        "mutation-level": "Complete",
        "test-projects": ["Project.Test.Unit/Project.Test.Unit.csproj"]
    }
}
```

**Comment by EvgeniKaidalov** [comment, marked as answer]
Looks like it's tied to so called "solution context" - when you run it from the solution directory, it analyzes the whole solution.
I've managed to filter tests by using "test-case-filter": "Category=Unit", requiring the attribute depending on the testing framework: https://github.com/Microsoft/vstest-docs/blob/main/docs/filter.md

**Reply by rouke-broersma** [reply]
Sorry I forgot to reply to this question. This is the correct way to do this yes!

**Comment by bNobo** [comment]
Instead of filtering you can simply execute the `dotnet stryker` command from the unit tests project folder. That way it will build and execute only the unit test project instead of building the entire solution and execute all tests.
